pkg/kubernetes/types.go

```diff
@@ -27,10 +29,28 @@ const (
 const (
 	// DefaultKubeEventResyncInterval is the default resync interval for k8s events
 	DefaultKubeEventResyncInterval = 30 * time.Second
+
+	// ProviderName is used for provider logging
+	ProviderName = "Kubernetes"
+
+	// Namespaces is a simple string to identify namespace resources
+	Namespaces = "Namespaces"
 )
 
 // Event is the combined type and actual object we received from Kubernetes
 type Event struct {
 	Type  EventType
 	Value interface{}
 }
+
+// InformerCollection stores the different Informers we keep for K8s resources
+type InformerCollection map[string]cache.SharedIndexInformer
```

Comment:
To prevent incorrect lookups from the wrong cache, I'd change it to how this is in `smi/types.go`
```
// InformerCollection is a struct of the Kubernetes informers used in OSM
type InformerCollection struct {
	Services       cache.SharedIndexInformer
	TrafficSplit   cache.SharedIndexInformer
	HTTPRouteGroup cache.SharedIndexInformer
	TCPRoute       cache.SharedIndexInformer
	TrafficTarget  cache.SharedIndexInformer
	Backpressure   cache.SharedIndexInformer
}

// CacheCollection is a struct of the Kubernetes caches used in OSM
type CacheCollection struct {
	Services       cache.Store
	TrafficSplit   cache.Store
	HTTPRouteGroup cache.Store
	TCPRoute       cache.Store
	TrafficTarget  cache.Store
	Backpressure   cache.Store
}
```